team.go

```diff
@@ -27,82 +27,176 @@ type ListTeamOptions struct {
+// recommended to use AddEscalationPolicyToTeamWithContext instead.
 func (c *Client) AddEscalationPolicyToTeam(teamID, epID string) error {
-	_, err := c.put(context.TODO(), "/teams/"+teamID+"/escalation_policies/"+epID, nil, nil)
+	return c.AddEscalationPolicyToTeamWithContext(context.Background(), teamID, epID)
+}
+
+// AddEscalationPolicyToTeamWithContext adds an escalation policy to a team.
+func (c *Client) AddEscalationPolicyToTeamWithContext(ctx context.Context, teamID, epID string) error {
+	_, err := c.put(ctx, "/teams/"+teamID+"/escalation_policies/"+epID, nil, nil)
 	return err
 }
 
-// RemoveUserFromTeam removes a user from a team.
+// RemoveUserFromTeam removes a user from a team. It's recommended to use
+// RemoveUserFromTeamWithContext instead.
 func (c *Client) RemoveUserFromTeam(teamID, userID string) error {
-	_, err := c.delete(context.TODO(), "/teams/"+teamID+"/users/"+userID)
+	return c.RemoveUserFromTeamWithContext(context.Background(), teamID, userID)
+}
+
+// RemoveUserFromTeamWithContext removes a user from a team.
+func (c *Client) RemoveUserFromTeamWithContext(ctx context.Context, teamID, userID string) error {
+	_, err := c.delete(ctx, "/teams/"+teamID+"/users/"+userID)
 	return err
 }
 
 // AddUserToTeam adds a user to a team.
 func (c *Client) AddUserToTeam(teamID, userID string) error {
-	_, err := c.put(context.TODO(), "/teams/"+teamID+"/users/"+userID, nil, nil)
+	return c.AddUserToTeamWithContext(context.Background(), AddUserToTeamOptions{TeamID: teamID, UserID: userID})
+}
+
+// TeamUserRole is a named type to represent the different Team Roles supported
+// by PagerDuty when adding a user to a team.
+//
+// For more info: https://support.pagerduty.com/docs/advanced-permissions#team-roles
+type TeamUserRole string
+
+const (
+	// TeamUserRoleObserver is the obesrver team role, which generally provides
+	// read-only access. They gain responder-level permissions on an incident if
+	// one is assigned to them.
+	TeamUserRoleObserver TeamUserRole = "observer"
+
+	// TeamUserRoleResponder is the responder team role, and they are given the
+	// same permissions as the observer plus the ability to respond to
+	// incidents, trigger incidents, and manage overrides.
+	TeamUserRoleResponder TeamUserRole = "responder"
+
+	// TeamUserRoleManager is the manager team role, and they are given the same
+	// permissions as the responder plus the ability to edit and delete the
+	// different resources owned by the team.
+	TeamUserRoleManager TeamUserRole = "manager"
+)
+
+// AddUserToTeamOptions is an option struct for the AddUserToTeamWithContext
+// method.
+type AddUserToTeamOptions struct {
+	TeamID string       `json:"-"`
+	UserID string       `json:"-"`
+	Role   TeamUserRole `json:"role,omitempty"`
+}
+
+// AddUserToTeamWithContext adds a user to a team.
+func (c *Client) AddUserToTeamWithContext(ctx context.Context, o AddUserToTeamOptions) error {
+	_, err := c.put(ctx, "/teams/"+o.TeamID+"/users/"+o.UserID, o, nil)
```

Comment:
The diff section above this comment was inspired by #262.

Reply:
I like this approach.